add "no-unused-vars" to ignore list in base configuration

ESLint's recommended rule "no-unused-vars" triggers if a YAML comment starts with the word "global", because ESLint takes this as file-level definition of global symbols, which is useless in YAML.

```yaml
# global symbols
data: [my, global, symbols]
```

gives

```none
  1:11  error  'symbols' is defined but never used  no-unused-vars
```

Thus, this rule should go into the base config next to "no-irregular-whitespace" and "spaced-comment".

--- a/.changeset/chilled-cobras-protect.md
+++ b/.changeset/chilled-cobras-protect.md
@@ -1,0 +1,5 @@
+---
+"eslint-plugin-yml": minor
+---
+
+disable `no-unused-vars` in base config

--- a/src/configs/base.ts
+++ b/src/configs/base.ts
@@ -7,6 +7,7 @@ export = {
       rules: {
         // ESLint core rules known to cause problems with YAML.
         "no-irregular-whitespace": "off",
+        "no-unused-vars": "off",
         "spaced-comment": "off",
       },
     },

